doc/docs/master/how-tos/enhanced-pipeline-workflow.md

```diff
@@ -0,0 +1,121 @@
+# Enhanced Pipeline Workflow
+
+The manual build steps mentioned in the [general pipeline workflow](../how-tos/working-with-data-and-pipelines.md) can become cumbersome during development cycles. Here we introduce two ways to automate these manual steps, namely build pipelines and the `--build` flag. 
+
+## Build Pipelines
+
+A build pipeline is a useful feature when iterating on the code in your pipeline. They allow you to bypass the Docker build process and submit your code directly to the pipeline (without having to re-build the Docker image). In essence, build pipelines automate Steps 1, 4, and 5 of the [general pipeline workflow](../how-tos/working-with-data-and-pipelines.md).
+
+!!! note
+      Build pipelines are not currently supported in Pachyderm Hub. 
+
+Functionally, the build pipeline uses a base Docker image that remains unchanged during the development process. The code is copied into this pipeline at runtime from a PFS repository. The build pipeline installs requirements and updates the code in the pipeline before running running it.
+
+Build pipelines require a modification to the pipeline spec's transform object with options for the following fields:
+
+- path: An optional string specifying where the source code is relative to the pipeline spec path (or the cwd if the pipeline is fed into pachctl via stdin.)
+- language: A string specifying what language builder to use (see below.) Only works with official builders.
+- image: A string specifying what builder image to use, if a non-official builder is desired.
+
+Below is a Python example of a build pipline.
+
+```json
+{
+  "pipeline": {
+    "name": "map"
+  },
+  "description": "A pipeline that tokenizes scraped pages and appends counts of words to corresponding files.",
+  "transform": {
+    "build": {
+      "language": "python",
+      "path": "./source"
+    }
+  },
+  "input": {
+    "pfs": {
+      "repo": "scraper",
+      "glob": "/*"
+    }
+  }
+}
+```
+
+A build pipeline can be submitted the same way as any other pipeline, for example:
+
+```bash
+pachctl update pipeline -f <pipeline name> 
+```
+
+When submitted, the following actions occur: 
+
+1. All files (code, etc.) are copied from the build path to a PFS repository, `<pipeline name>_build`, which we can think of as the source code repository. In the case above, everything in `./source` would be copied to to the PFS `<pipeline name>_build` repository.
+
+1. Starts a pipeline called `<pipeline name>_build` that reads the files uploaded and runs the `build.sh` script to pull dependencies and compile any requirements from the source code. 
+
+1. Creates (or updates if it already exists) the running pipeline `<pipeline name>` with the the newly creates source files and built assets. 
+
+!!! note
+      You can optionally specify a `.pachignore` file to prevent certain files from getting pushed to this repo. 
```

Comment:
We should note here or somewhere else that `.pachignore` uses [ohmyglob](https://github.com/pachyderm/ohmyglob) entries for ignores

Reply:
Updated in most recent commit.